packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -328,9 +336,10 @@ class HotRunner extends ResidentRunner {
         fullRestart: fullRestart,
         projectRootPath: projectRootPath,
         pathToReload: getReloadPath(fullRestart: fullRestart),
-      );
-      if (!result)
-        return false;
+      ));
+    }
+    if (results.any((bool result) => !result)) {
```

Comment:
Might want a comment on this conditional: I had to read it several times to realize what it's doing.